Tests/ImportDumpTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+use Cybex\Protector\Exceptions\FailedShellCommandException;
+use Cybex\Protector\Exceptions\FileNotFoundException;
+use Cybex\Protector\Exceptions\InvalidConnectionException;
+use Cybex\Protector\Exceptions\InvalidEnvironmentException;
+use Cybex\Protector\Protector;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Facades\Storage;
+
+class ImportDumpTest extends BaseTest
+{
+    /**
+     *  Protector instance.
+     */
+    protected Protector $protector;
+
+    protected string $baseDirectory;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Config::set('protector.baseDirectory', 'protector');
+
+        $this->protector     = app('protector');
+        $this->disk          = Storage::disk('local');
+        $this->baseDirectory = Config::get('protector.baseDirectory');
+        $this->filePath      = sprintf('%s/dump.sql', $this->baseDirectory);
+    }
+
+    /**
+     * @test
+     */
+    public function failOnProductionEnvironment()
+    {
+        $this->app->detectEnvironment(fn() => 'production');
+
+        $this->expectException(InvalidEnvironmentException::class);
+        $this->protector->importDump($this->filePath, ['allow-production' => false]);
```

Comment:
I need to explicitly set "allow-production" to false, because while importing, we check if the Environment is on production and the "allow-production" - option is set to false. Otherwise no exception is thrown at the moment.

Reply:
Fixed.